### DISCUSSION TITLE
How to use `readdir` in Vercel deployments?

### DISCUSSION THREAD

**Discussion by piet-maier** [vercel/community#1628] [Help]
### Summary

I have a Nuxt 3 project and a bunch of images inside various sub-folders of my `/public/images` directory which I want to display as an image gallery on my website. Since I don't want to hard-code the image paths, I set up an endpoint that uses a simple `readdir` call to read the contents of the folder and returns an array containing all image paths.

This works on my machine, on the development server as well as when building the project, but it doesn't work when deployed on Vercel. The error I get tells me that there is no directory `/var/task/public/images` that could be read. I found various posts on GitHub and StackOverflow describing the same or similar problems, but no clear solution.

### Additional information

Simple Code Example:

``` JavaScript
import { readddir } from "node:fs/promises";
import { join } from "node:path";

export default defineEventHandler(async (_event) => {
  const path = join(process.cwd(), "/public/images");

  return images = await readdir(path);
});
```

**Comment by StarrKiss** [comment]
Realized i have the same/similar problem too,  see here: https://github.com/orgs/vercel/discussions/3083
